### PR TITLE
FEM-1429

### PR DESCRIPTION
**Pull request by giladna**
Support FileId from MediaSource instead of application json
for tvpapi and phoenix analytics plugins

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java [hunk 1]
```diff
@@ -250,13 +246,18 @@ public void run() {
      * @param eventType - Enum stating the event type to send
      */
     protected void sendAnalyticsEvent(final PhoenixActionType eventType) {
+        if ("".equals(getBaseUrl())) {
```
Comment (OrenMe) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: I think we should only do this on init and config update

File: playkit/src/main/java/com/kaltura/playkit/plugins/TVPAPIAnalyticsPlugin.java [hunk 2]
```diff
@@ -44,8 +43,11 @@ public void warmUp(Context context) {
      */
     @Override
     protected void sendAnalyticsEvent(final PhoenixActionType eventType){
-        String fileId = pluginConfig.has("fileId")? pluginConfig.getAsJsonPrimitive("fileId").getAsString():"000000";
-        String baseUrl = pluginConfig.has("baseUrl")? pluginConfig.getAsJsonPrimitive("baseUrl").getAsString():"http://tvpapi-preprod.ott.kaltura.com/v3_9/gateways/jsonpostgw.aspx?";
+        if ("".equals(getBaseUrl())) {
```
Comment (OrenMe) [inline playkit/src/main/java/com/kaltura/playkit/plugins/TVPAPIAnalyticsPlugin.java]: Same as in before, should do it on init and config update

File: playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java [hunk 3]
```diff
@@ -94,16 +92,10 @@ protected void onUpdateMedia(PKMediaConfig mediaConfig) {
...
-            fileId = pluginConfig.getAsJsonPrimitive("fileId").getAsString();
-        } else {
-            log.e("Error PhoenixAnalytics fileId was not set");
-            fileId = "0";
-        }
         if (Utils.isJsonObjectValueValid(pluginConfig, "baseUrl")) {
             baseUrl = pluginConfig.getAsJsonPrimitive("baseUrl").getAsString();
         } else {
-            baseUrl = DEFAULT_BASE_URL;
+            baseUrl = "";
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: Please use null as a marker for "no value", not an empty string.

File: playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java (hunk 1 above)
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: Agreed.

File: playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java [hunk 4]
```diff
@@ -272,4 +273,15 @@ public void onComplete(ResponseElement response) {
         requestsExecutor.queue(requestBuilder.build());
     }
 
+    public String getFileId() {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: Isn't `fileId` an integer?

File: playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java [hunk 5]
```diff
@@ -272,4 +273,15 @@ public void onComplete(ResponseElement response) {
...
+        return (fileId != null) ? fileId : "000000";
+    }
+
+    public String getBaseUrl() {
+        if (baseUrl != null) {
+            return baseUrl;
+        } else {
+            return "";
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: Same -- return null as no value.

Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: just return what I have null will check later

File: playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java (hunk 4 above)
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: yes as String

File: playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java (hunk 3 above)
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/plugins/PhoenixAnalyticsPlugin.java]: OK

File: playkit/src/main/java/com/kaltura/playkit/plugins/TVPAPIAnalyticsPlugin.java (hunk 2 above)
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/plugins/TVPAPIAnalyticsPlugin.java]: OK just ask for null for preventing sending it

File: playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPIInitObject.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+    public static final String SITE_GUID   = "SiteGuid";
+    public static final String API_USER    = "ApiUser";
+    public static final String DOMAIN_ID   = "DomainID";
+    public static final String UDID_KEY    = "UDID";
+    public static final String API_PASS    = "ApiPass";
+    public static final String PLATFORM    = "Platform";
+    public static final String LOCALE      = "Locale";
+
+    private int SiteGuid;
```
Comment (gal-orlanczyk) [inline playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPIInitObject.java]: @noamtamim variables should be lower case

File: playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPIInitObject.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    private String ApiUser;
+    private int DomainId;
+    private String UDID;
```
Comment (gal-orlanczyk) [inline playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPIInitObject.java]: @noamtamim should be uuid no?

File: playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPILocale.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    public static final String LOACALE_LANGUAGE   = "LocaleLanguage";
+
+
+    // "Locale":
+    private String LocaleUserState;
+    private String LocaleCountry;
+    private String LocaleDevice;
+    private String LocaleLanguage;
+
+    public TVPAPILocale(String loacalUserState, String localeCountry, String localeDevice, String localeLanguage) {
```
Comment (gal-orlanczyk) [inline playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPILocale.java]: @noamtamim typo in `localUserState`

File: playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPIInitObject.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * Created by gilad.nadav on 22/06/2017.
+ */
+
+public class TVPAPIInitObject {
```
Comment (gal-orlanczyk) [inline playkit/src/main/java/com/kaltura/playkit/plugins/configs/TVPAPIInitObject.java]: @noamtamim missing token property.

**Pull request merged**